tests/components/anthemav/test_media_player.py

```diff
@@ -0,0 +1,69 @@
+"""Test the Anthem A/V Receivers config flow."""
+from unittest.mock import AsyncMock
+
+import pytest
+
+from homeassistant.components.anthemav.const import ANTHEMAV_UDATE_SIGNAL
+from homeassistant.components.media_player.const import (
+    ATTR_APP_NAME,
+    ATTR_INPUT_SOURCE,
+    ATTR_INPUT_SOURCE_LIST,
+    ATTR_MEDIA_TITLE,
+    ATTR_MEDIA_VOLUME_MUTED,
+)
+from homeassistant.components.siren.const import ATTR_VOLUME_LEVEL
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize(
+    "entity_id,entity_name",
+    [
+        ("media_player.anthem_av", "Anthem AV"),
+        ("media_player.anthem_av_zone_2", "Anthem AV zone 2"),
+    ],
+)
+async def test_zones_loaded(
+    hass: HomeAssistant,
+    init_integration: MockConfigEntry,
+    entity_id: str,
+    entity_name: str,
+) -> None:
+    """Test zones are loaded."""
+
+    states = hass.states.get(entity_id)
+
+    assert states
+    assert states.state == STATE_OFF
+    assert states.name == entity_name
+
+
+async def test_update_states_zone1(
+    hass: HomeAssistant, init_integration: MockConfigEntry, mock_anthemav: AsyncMock
+) -> None:
+    """Test zone states are updated."""
+
+    mock_zone = mock_anthemav.protocol.zones[1]
+
+    mock_zone.power = True
+    mock_zone.mute = True
+    mock_zone.volume_as_percentage = 42
+    mock_zone.input_name = "TEST INPUT"
+    mock_zone.input_format = "2.0 PCM"
+    mock_anthemav.protocol.input_list = ["TEST INPUT", "INPUT 2"]
+
+    async_dispatcher_send(hass, f"{ANTHEMAV_UDATE_SIGNAL}_{init_integration.entry_id}")
```

Comment:
Please patch the library to get access to the callback to update state.

Reply:
I added a fixture to get the callback function. Not sure if that's the best way to do it.